### PR TITLE
DOCS new guidance page on how to upgrade a project

### PR DESCRIPTION
**Pull request by brynwhyman**

### CODE REVIEW

File: docs/en/03_Upgrading/01_Keeping_projects_up_to_date.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+Assuming your project is using one of the [supported recipes](../00_Getting_Started/05_Recipes.md), these will likely take more time as the APIs may change between minor and major releases. For small sites it's possible for minor upgrade to take a day of work, and major upgrades could take several days. Of course this can widely differ depending on each project.
+
+To upgrade your code, open the root `composer.json` file. Find the lines that reference the recipes, like  `silverstripe/recipe-cms` and change the referenced versions to what has been reference in the changelog (as well as any other modules that have a new version).
+
+For example, assuming that you are currently on version `~4.8.0@stable`, if you wish to upgrade to 4.9.0 you will need to modify your `composer.json` file to explicitly specify the new release branch, here `~4.9.0`:
+
+```json
+...
+"require": {
+"silverstripe/recipe-cms": "~4.9.0"
```
Comment (michalkleiner) [inline docs/en/03_Upgrading/01_Keeping_projects_up_to_date.md:33]:
```suggestion
    "silverstripe/recipe-cms": "~4.9.0"
```

File: docs/en/03_Upgrading/01_Keeping_projects_up_to_date.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+},
+...
+```
+
+You now need to pull in new dependencies and commit the lock file:
+
+```bash
+composer update
+git commit composer.* -m "Upgrading to recipe 4.9.0"
```
Comment (michalkleiner) [inline docs/en/03_Upgrading/01_Keeping_projects_up_to_date.md:42]:
```suggestion
git commit composer.* -m "Upgrade to recipe 4.9.0"
```

File: docs/en/03_Upgrading/01_Keeping_projects_up_to_date.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+```
+
+Push this commit to your remote repository, and continue with your deployment workflow.
+
+## Cherrypicking the upgrades
+
+If you like to only upgrade the recipe modules, you can cherry pick what is upgraded using this syntax:
+
+	composer update silverstripe/recipe-cms
```
Comment (michalkleiner) [inline docs/en/03_Upgrading/01_Keeping_projects_up_to_date.md:51]:
```suggestion
`composer update silverstripe/recipe-cms`
```

File: docs/en/03_Upgrading/01_Keeping_projects_up_to_date.md (hunk 1 above)
Comment (brynwhyman) [reply docs/en/03_Upgrading/01_Keeping_projects_up_to_date.md]: Thanks for the review, updated this.

File: docs/en/03_Upgrading/01_Keeping_projects_up_to_date.md (hunk 3 above)
Comment (brynwhyman) [reply docs/en/03_Upgrading/01_Keeping_projects_up_to_date.md]: Updated

File: docs/en/03_Upgrading/01_Keeping_projects_up_to_date.md (hunk 2 above)
Comment (brynwhyman) [reply docs/en/03_Upgrading/01_Keeping_projects_up_to_date.md]: Updated

**Review by emteknetnz — APPROVED** [review]

**Comment by emteknetnz** [comment]
@brynwhyman Thank you for your contribution ;-)

**Pull request merged**